src/components/date/date.component.js

```diff
@@ -315,9 +313,8 @@ class BaseDateInput extends React.Component {
   handleVisibleInputChange = (ev) => {
     const { disabled, readOnly } = this.props;
     const value = ev.target.value.formattedValue || ev.target.value;
-    const dateWithSlashes = DateHelper.sanitizeDateInput(value);
     const isValidDate = DateHelper.isValidDate({
-      value: dateWithSlashes,
+      value,
```

Comment:
I think this change will prevent some valid input formats from setting the selectedDate in the picker

Reply:
it would prevent setting a date in a format not specified by an user, if they decide to add their own valid formats

Reply:
another issue with that is when a user is setting their own list of valid formats that is not containing the one with slashes, this method would throw an error for correct dates!

Reply:
Yeah but it prevents previously reformatted values from being formatted as well, I think it would be better to add a guard against attempting to pass an `invalid date` to the picker if the format is custom etc

Reply:
moment already does it for us, there is no need to convert dots, hyphens and spaces into slashes